### ISSUE TITLE
Temporary table objects persist across sessions in Oracle versions older than 18c

### ISSUE THREAD

**Issue by NathanFarmer** [fivetran/great_expectations#2861]
**Bug description:**
Temporary tables that are created when running `batch = context.get_batch(batch_kwargs, suite)` persist across Oracle sessions. The data itself does not persist, but the table object will remain.

**To reproduce:**
1. `great_expectations suite edit [expectation_suite_name]`
2. Run Jupyter Notebook cell 1
3. Log out, shut down Jupyter server, end Oracle session
4. Restart Oracle session and find a table that starts with "GE_TMP_"

**Expected behavior:**
Once an Oracle session has ended, it is expected that temporary tables would no longer exist in the database. While this is not the direction that Oracle took in the first ~39 years of its database's existence, the Great Expectations codebase would be greatly simplified by treating temporary tables as transient objects, as they are in the vast majority of other db dialects.

**Environment:**
 - Operating System: MacOS Big Sur 11.4
 - Great Expectations Version: 13.19
 - Oracle Versions: Tested on 12.1.0.2.0 Enterprise

**Additional context:**
This particular quirk of Oracle was handled with the introduction of [Private Temporary Tables](https://oracle-base.com/articles/18c/private-temporary-tables-18c) in Oracle 18c.

The problem with that as far as Great Expectations is concerned, is that Oracle 18c, 19c, and 21c seem to have had relatively low adoption rates to-date. The only data I could find on this can be found in [this article](https://www.brightworkresearch.com/what-percentage-of-oracle-customers-are-on-what-oracle-database-version/#:~:text=Overall%2C%20roughly%20an%20insignificant%206,The%20question%20is%20why.), where the author estimates that as of June 2019 only 6% of Oracle customers had upgraded to Oracle 12c and an insignificant percentage of users had migrated to 18c or 19c. 

**Proposed partial solution:**
I have opened PR [#2862](https://github.com/great-expectations/great_expectations/pull/2862) to address this issue in Oracle versions 18c, 19c, and 21c through the use of `PRIVATE TEMPORARY` tables. That still leaves the issue of Oracle versions 12c and older which is in use by the vast majority of Oracle customers.

I am handling the "garbage collection" myself through periodically running the PL/SQL block below:
```SQLPL
BEGIN
  FOR c IN ( SELECT table_name FROM user_tables WHERE table_name LIKE 'GE_TMP_%' )
  LOOP
    EXECUTE IMMEDIATE 'DROP TABLE ' || c.table_name;
  END LOOP;
END;
```

I looked into adding the function below somewhere to clean up the table that was recently created, but I'm not sure where it would go or how it would work given the persistent nature of the data context.

Oracle `GLOBAL TEMPORARY` tables can only be dropped once they are empty and therefore a truncate is required if you have not entered into a new Oracle session.

```python
def drop_temporary_table(self, table_name):
    """Drop temporary tables for dialects where temporary tables are not transient objects.
       Example: Oracle temp tables before 18c only had transient data, not transient table definitions."""
    engine_dialect = self.sql_engine_dialect.name.lower()
    # handle cases where dialect.name.lower() returns a byte string (e.g. databricks)
    if isinstance(engine_dialect, bytes):
        engine_dialect = str(engine_dialect, "utf-8")
        
    if engine_dialect == "oracle":
        # oracle temp tables cannot be dropped if data persists so we must truncate first
        stmt_1 = "TRUNCATE TABLE {table_name}".format(
            table_name=table_name
        )
        stmt_2 = "DROP TABLE {table_name}".format(
            table_name=table_name
        )

        self.engine.execute(stmt_1)
        self.engine.execute(stmt_2)
```

**Comment by eugmandel** [comment]
@NathanFarmer I saw that you created a PR for this issue - thank you!

**Comment by NathanFarmer** [comment]
This issue was addressed by PR #2862 for Oracle versions 18c and newer. This issue is still open for versions older than 18c.

**Comment by AFineDayFor** [comment]
closing, we can't prioritize oracle at this time.
